src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -129,12 +126,12 @@ export class CourseExercisesComponent implements OnInit, OnChanges, OnDestroy, A
             this.courseId = parseInt(params['courseId'], 10);
         });
 
-        this.course = this.courseCalculationService.getCourse(this.courseId);
+        this.course = this.courseStorageService.getCourse(this.courseId);
         this.onCourseLoad();
 
-        this.courseUpdatesSubscription = this.courseService.getCourseUpdates(this.courseId).subscribe((course: Course) => {
-            this.courseCalculationService.updateCourse(course);
-            this.course = this.courseCalculationService.getCourse(this.courseId);
+        this.courseUpdatesSubscription = this.courseStorageService.subscribeToCourseUpdates(this.courseId).subscribe((course: Course) => {
+            this.courseStorageService.updateCourse(course);
+            this.course = this.courseStorageService.getCourse(this.courseId);
```

Comment:
Sorry to break the approval streak, but this code still seems a bit off to me.

You already receive a course update by the courseStorageService, just to pass the course back to the service and then load the same object from the service again. Could this lead to an infinite recursion?
```suggestion
            this.course = course;
```